src/Table/Table.tsx

```diff
@@ -0,0 +1,977 @@
+            } else {
+                result = (firstItem[columnName] - secondItem[columnName]);
+            }
+        } else {
+            if (isNaN(secondItem[columnName]) && isNaN(firstItem[columnName])) {
+                result = String(secondItem[columnName]).localeCompare(String(firstItem[columnName]), ["sw", "en"], { sensitivity: "base", ignorePunctuation: true });
+            } else {
+                result = (secondItem[columnName] - firstItem[columnName]);
+            }
+        }
+        return result;
+
+    });
+    return sortedItems;
+}
+
+/**
+ * search text in array of table row
+ * @param items the array of table rows
+ * @param keyword The keyword to search in the array
+ * @param searchFields the target field to search
+ */
+function searchTextInArray(items: Array<TableRow>, keyword: string, searchFields: Array<string>): Array<TableRow> {
+    return [...items].filter((row: TableRow) => {
+        if (keyword.trim().length === 0 || searchFields.length === 0) {
+            return true;
+        }
+        return (
+            searchFields.some((searchColumn: string) => {
+                let result: boolean = false;
+                const searchField: string = searchColumn;
+                const regEx = new RegExp(keyword, "gi");
+                if (row[searchField] === null || row[searchField] === undefined) {
+                    result = false;
+                } else if (typeof row[searchField] === "string") {
+                    result = row[searchField].search(regEx) > -1;
+                } else if (typeof row[searchField] === "number") {
+                    result = String(row[searchField]).search(regEx) !== -1;
+                }
+                return result;
+            })
+        );
+    });
+}
+
+interface ActionColumnProps {
+    actionLinks?: Array<ActionLinkItem>;
+    onActionDropped?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+    primaryActionButton?: PrimaryActionButton;
+    selectedRow: TableRow;
+    tableRef: React.RefObject<HTMLTableElement>;
+}
+
+const ActionColumn: React.FunctionComponent<ActionColumnProps> = (props: ActionColumnProps) => {
+    const btnPrimaryRandomIds = randomId("btn");
+    const [dropup, setDropup] = React.useState<boolean>(false);
+    const actionRef: React.RefObject<HTMLDivElement> = React.createRef<HTMLDivElement>();
+
+    function getActionColumnClasses() {
+        let className: string = "dropdown-content";
+        if (props.selectedRow?.actionsDropdownDropped) {
+            className += " active";
+        }
+
+        if (dropup) {
+            className += " dropup";
+        }
+
+        return className;
+    }
+
+    return (
+        <div className="action-column">
+            {props.primaryActionButton &&
+                <button
+                    id={btnPrimaryRandomIds}
+                    type="button"
+                    className="btn btn-outline-primary btn-sm"
```

Comment:
same for className

Reply:
I disagree with this too. Considering each row renders this , i don't see a use for a className. If the user want to customise this , he can rely on his custom className and the last action column, which is one thing the buttons have in common. I don't foresee a situation where the user customise one button leaving the rest.